### PR TITLE
Features/tsc merkle proofs

### PR DESCRIPTION
**Pull request by AustEcon**

### CODE REVIEW

File: electrumx/server/session.py [hunk 1]
```diff
@@ -643,7 +643,7 @@ def extra_cost(self, session):
             return 0
         return sum((group.cost() - session.cost) * group.weight for group in groups)
 
-    async def _merkle_branch(self, height, tx_hashes, tx_pos):
+    async def _merkle_branch(self, height, tx_hashes, tx_pos, return_root=False):
```
Comment (kyuupichan) [inline electrumx/server/session.py:646]: I don't like functions returning differently-sized tuples based on how they're called.  Please return the root always now, and update the former callers to expect and ignore it.

**Review by kyuupichan — CHANGES_REQUESTED** [review]
The new method also needs documenting in the docs/ directory.

Is adding tests too difficult?  If it is, it's not a problem.

File: electrumx/server/session.py (hunk 1 above)
Comment (AustEcon) [reply electrumx/server/session.py:646]: Okay no problem. No adding tests is okay - I still have local changes I haven't pushed. The main niggle is with adding asterixes in place of "duplicate hashes" in the branch but I will see what you think after I push the next commit(s)...

**Review by Choicezee01 — APPROVED** [review]

**Review by Choicezee01 — CHANGES_REQUESTED** [review]
34xp4vRoCGJym3xR7yCVPFHoCNxv4Twseo

**Pull request merged**